Mathlib/Algebra/Ring/CentroidHom.lean

```diff
@@ -432,6 +433,54 @@ instance : DistribMulAction M (CentroidHom α) :=
 instance : Module R (CentroidHom α) :=
   toEnd_injective.module R (toEndRingHom α).toAddMonoidHom toEnd_smul
 
+/-!
+The following instances show that `α` is an algebra over `CentroidHom α`
+-/
+
+/-- The tautological action by `CentroidHom α` on `α`.
+
+This generalizes `Function.End.applyMulAction`. -/
+instance applyModule : Module (CentroidHom α) α where
+  smul T a := T a
+  add_smul _ _ _ := rfl
+  zero_smul _ := rfl
+  one_smul _ := rfl
+  mul_smul _ _ _:= rfl
+  smul_zero := map_zero
+  smul_add := map_add
+
+@[simp]
+lemma smul_def (T : CentroidHom α) (a : α) : T • a = T a := rfl
+
+instance : SMulCommClass (CentroidHom α) α α where
+  smul_comm _ _ _ := map_mul_left _ _ _
+
+instance : SMulCommClass α (CentroidHom α) α := SMulCommClass.symm _ _ _
+
+instance : IsScalarTower (CentroidHom α) α α where
+  smul_assoc _ _ _ := (map_mul_right _ _ _).symm
+
+/-!
+Let `α` be an algebra over `R`, such that the canonical ring homomorphism of `R` into
+`CentroidHom α` lies in the center of `CentroidHom α`. Then `CentroidHom α` is an algebra over `R`
+-/
+
+variable {R : Type*}
+
+variable [CommSemiring R]
+variable [Module R α] [SMulCommClass R α α] [IsScalarTower R α α]
+
+/-- The natural ring homomorphism from `R` into `CentroidHom α`.
+
+This is a stronger version of `Module.toAddMonoidEnd`. -/
+@[simps! apply_toFun]
+def _root_.Module.toCentroidHom : R →+* CentroidHom α := RingHom.smulOneHom
+
+open Module in
+/-- `CentroidHom α` as an algebra over `R`. -/
+abbrev ringToAlg (h : ∀ (r : R) (T : CentroidHom α), toCentroidHom r * T = T * toCentroidHom r) :
+    Algebra R (CentroidHom α) := toCentroidHom.toAlgebra' h
```

Comment:
I'm not fully convinced that this definition is worth having, since it is only a little shorter than writing `Module.toCentroidHom.toAlgebra' h`.

Do you have a use-case in mind where this shorter spelling makes much difference? If not, I'd be inclined to remove it.

(The same does not apply to `toCentroidHom` above, as that's following an existing pattern)

Reply:
@eric-wieser  I can remove `ringToAlg` - but then it feels to me like this PR isn't quite reaching it's natural conclusion.

How would you feel about:
```
open Module in
/-- `CentroidHom α` as an algebra over `R`. -/
example (h : ∀ (r : R) (T : CentroidHom α), toCentroidHom r * T = T * toCentroidHom r) :
    Algebra R (CentroidHom α) := toCentroidHom.toAlgebra' h
```
?

Reply:
Sure, that seems harmless. Did you have a result you were intending to build on top of this in a later PR?

Reply:
@eric-wieser not specifically, but when `α` is a module over `R`, the homomorphisms are usually taken to be `R`-linear so I though it important to indicate how to get from our ring definition to the module/algebra situation.